app/scripts/lib/segment.js

```diff
@@ -82,10 +83,10 @@ export const createSegmentMock = (flushAt = SEGMENT_FLUSH_AT) => {
 };
 
 export const segment =
-  !SEGMENT_WRITE_KEY || (isDevEnvironment && !SEGMENT_HOST)
-    ? createSegmentMock(SEGMENT_FLUSH_AT, SEGMENT_FLUSH_INTERVAL)
-    : new Analytics(SEGMENT_WRITE_KEY, {
+  SEGMENT_WRITE_KEY && !isDevEnvironment
```

Comment:
i could be wrong here, but wouldn't this result in the mock being used in development mode? Which means now even with SEGMENT_HOST set events don't make it to segment. METAMASK_DEBUG is true in dev mode. I think we only want to check against IN_TEST

Reply:
@brad-decker 

Oh, I'm just now realizing that the METAMASK_DEBUG flag can be tied with dev mode and we want to negate this.

Now I see the purpose of `!SEGMENT_HOST` in the original code is a flag to allow segment in dev mode for the case you just mentioned.

So we're back at what @Gudahtt  mentioned in the above comment and what you are mentioning now - we could use
```SEGMENT_WRITE_KEY && !process.env.IN_TEST```

Reply:
Yeah, I think that makes sense.

Thinking about this in a different way.... what we want is to make `SEGMENT_WRITE_KEY` determine if Segment is used locally, but use something else for determining when Segment is used on CI, because `SEGMENT_WRITE_KEY` is always set on CI.

Right now there are two CI build types, prod and testing. Segment is enabled on prod but not on testing. So this preserves that behavior.

The one problem right now is that it's impossible to enable segment with a test build now. Maybe this doesn't matter, but it doesn't seem ideal.

One idea would be to use the condition `SEGMENT_WRITE_KEY && (process.env.METAMASK_DEBUG || !process.env.IN_TEST)` instead. i.e. `SEGMENT_WRITE_KEY &&  isProdOrDevMode`. That would let us use segment locally for a _dev_ test build, but still not for a non-dev test build. That would probably be good enough for all circumstances though.

Reply:
Oh darn, this breaks the metrics tests. We _do_ need to enable Segment on CI test builds. Just the host is set to localhost in that environment. I forgot about that. Hmmm....

Reply:
Right, so when I said "Currently this would send e2e metrics to Segment, which definitely isn't what we want.", I was wrong. That is what we want, because we're intercepting those requests during e2e test runs using the new e2e test mock server, so they will never reach Segment. Sorry about that!

So maybe the condition `SEGMENT_WRITE_KEY` is all we need. We want Segment to always be enabled on CI, because the e2e test mocks will intercept the requests anyway. And locally the presence of that key is enough.